packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -351,8 +358,16 @@ export const GridRootStyles = styled('div', {
       padding: 0,
       alignItems: 'stretch',
     },
-    [`& .${gridClasses.withBorder}`]: {
-      borderRight: `1px solid ${borderColor}`,
+    [`.${gridClasses.withBorderColor}`]: {
+      borderColor,
+    },
```

Comment:
Yes, I did consider it, but it breaks this demo: https://mui.com/x/react-data-grid/style/#using-the-sx-prop
This means that you won't be able to override the border of the grid with:
```tsx
<DataGrid
  sx={{ borderColor: 'primary.light' }}
/>
```
Because the default border color has higher specificity:
```tsx
[`&.${gridClasses.withBorderColor}`]: {
  borderColor,
},
```

So I decided to not to do this and allow to override grid external border color with `sx` instead.